File: 240322/정수 n개의 합 2/sum-of-n-integers-2.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int answer = -1;
        int n = sc.nextInt();
        int k = sc.nextInt();

        int[] arr = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            arr[i] = sc.nextInt();
        }

        int[] sum = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            sum[i] = sum[i-1] + arr[i];
        }

        for (int end = k; end <= n; end++) {
            int tmp = end - k;

            int result = sum[end] - sum[tmp];
            answer = Math.max(answer, result);
        }
        System.out.print(answer);
    }
}

File: 240409/정수 n개의 합 2/sum-of-n-integers-2.kt
import java.util.*

val sc = Scanner(System.`in`)

val n = sc.nextInt()
val k = sc.nextInt()
val arr = IntArray(n + 1)

fun main() {
    for (i in 1..n) {
        arr[i] = sc.nextInt()
    }

    val sums = IntArray(n + 1)
    sums[1] = arr[1]
    for(i in 2..n) {
        sums[i] = sums[i - 1] + arr[i]
    }

    var answer = 0
    for (i in k..n) {
        val s = i - k + 1
        val e = i

        val sum = sums[e] - sums[s - 1]
        answer = Math.max(answer, sum)
    }

    print(answer)
}